Update PR template (#618)
JCSDA/spack-stack: .github/pull_request_template.md

### Summary

Describe the changes made in this PR and why they are needed.

### Testing

Describe the testing done for this PR.

### Applications affected

> List all known applications (UFS WM, JEDI, SRW, etc.) intentionally or unintentionally affected by this PR.

Reviewer: just one question about wording here: if an app or system is unintentionally affected by a pr, would the PR author be aware of that at the time of creating the PR?

Reviewer 2: See above

> List all known applications (UFS WM, JEDI, SRW, etc.) intentionally or unintentionally affected by this PR.

Reviewer 2: How do I know that my PR affects something unintentionally?

Author: I kept this language from the JCSDA template because I like that it sets the expectation that PR submitters are responsible for not breaking things that aren't relevant to their application of interest (and for that matter, it might prompt them to think harder about the impacts of their changes).

Reviewer 2: Interesting twist that you have to "defend" the PR template against folks from that very organization ;-) Let's keep that. But we don't have (and don't use) then "waiting for another PR" label for spack-stack, so maybe we can change the wording on just that line and be good with the PR?

### Systems affected

> List all systems intentionally or unintentionally affected by this PR.

Reviewer 2: Same here?

### Dependencies

> If there are PRs that need to be merged before or along with this one, please add "waiting for another PR" label and link to the relevant issues/PRs.

Reviewer 2: This label does not exist in the spack-stack repository. How about this:
```suggestion
If there are PRs that need to be merged before or along with this one, please add "- [ ] waiting on LINK_TO_PR" for each of them
```

Reviewer 2: @AlexanderRichert-NOAA If you can make this one change, then we can merge this PR and be done with it. Thanks!
